Release Candidate 2 for Version 0.4

mac-can opened the pull request.

mac-can committed:
Fix a bug with option `--trace`

Relates to [MACCAN-229]

mac-can committed:
Update `Makefile` for newer `cppcheck`er

Relates to [MACCAN-228]

mac-can committed:
Update `Makefile`s

Relates to [MACCAN-228]

mac-can committed:
Update the trial program

Relates to [MACCAN-229]

mac-can committed:
Update Python wrapper to v0.3.2

Relates to [MACCAN-229]

mac-can committed:
Update `Makefile`s

Relates to [MACCAN-228]

mac-can committed:
Update README.md

mac-can committed:
Set version number for development

Relates to [MACCAN-228]

mac-can committed:
Add CAN/IPC sources  (rev. 1433)

Relates to [MACCAN-233]

mac-can committed:
Integrate CAN/IPC into testing (rev. 1430)

Relates to [MACCAN-233]

mac-can committed:
Add CAN-to-Ethernet utility `can_port`

Relates to [MACCAN-233]

mac-can committed:
Update the example programs

Relates to [MACCAN-233]

mac-can committed:
Update README.md

Relates to [MACCAN-228]

uv-software commented on README.md:
```diff
@@ -128,11 +128,11 @@ Type `can_test --help` to display all program options.
+- macOS Ventura (13.7.3) on a MacBook Pro (2019)
 - Apple clang version 14.0.3 (clang-1403.0.22.14.1)
 
 #### Debian 12.9 ("bookworm")
```
Update to `Debian 6.1.124-1 (2025-01-12) x86_64 GNU/Linux`

Commit:
Update README.md

Relates to [MACCAN-228]

uv-software reviewed (approved):
Findings fixed on Linux machine

mac-can committed:
Add any key

Relates to [MACCAN-233]

Commit:
Fix a bug with 64-bit value byte order on Linux

Relates to [MACCAN-233]

Commit:
Rework CAN/IPC event handler for Linux version

- if the transmit queue is full, retry until the message is sent or a time-out occurred
- this could not happen with the macOS version (no transmit queue implemented)

Relates to [MACCAN-233]

mac-can committed:
Merge branch 'moonshake' into development
